src/libexpr/include/nix/expr/value.hh

```diff
@@ -35,6 +36,7 @@ typedef enum {
     tBool,
     tNull,
     tFloat,
+    tFailed,
```

Comment:
@xokdvium does this look ok?

Reply:
Thought: this could be a thunk instead, pointing to a pseudo-`Expr` that throws the error.
I don't think that change is required, but it would free up this scarce `InternalType` entry.